src/index.tsx

```diff
@@ -69,20 +69,22 @@ export function applyContainerQuery<T>(
 
     private cqCore: ContainerQueryCore | null = null;
 
-    constructor(props: T) {
+    constructor(props: T & EnhancedProps) {
       super(props);
 
       this.state = {
         params: initialSize
-          ? matchQueries(query)(initialSize)
+          ? matchQueries(props.transformQuery && props.transformQuery(query) || query)(initialSize)
           : {},
       };
     }
 
     componentDidMount() {
-      this.cqCore = new ContainerQueryCore(query, (params) => {
-        this.setState({params});
-      });
+      const { transformQuery } = this.props;
+      this.cqCore = new ContainerQueryCore(
+        transformQuery && transformQuery(query) || query,
```

Comment:
This where things starts to get tricky. 

1. If we allow to pass `transformQuery` in `render`, we have to anticipate the possibility that developers will pass a new `transformQuery` in a later time. This will requires us to implement a `componentWillReceiveProps` to handle this case. 
2. Even if `transformQuery` won't change, the logic inside might be dynamic, which leads to `query` passed to `ContainerQueryCore` becomes dynamic.

This will make the HOC implementation more complex. Which is the biggest reason I'm not very comfortable moving forward here. (I also noticed I didn't handle the updating of `query` props in `<ContainerQuery/>` API, oops 🤭)

On the other hand, all functionalities we want to achieve with `transformQuery`, we can achieve using `<ContainerQuery/>` API. If you feel it's cumbersome having to write `<ContainerQuery/>` every time, you can create your own version of HOC wrapping around `<ContainerQuery/>`.

Reply:
Yes I followed your example from ContainerQuery haha. It does need componentWillReceiveProps.

Thanks for the review, I'm a typescript newbie so it's extremely valuable :) I'll fix everything soon.

I created this PR for cases where render props won't be as performant as a HOC, but dynamic behavior is required. The penalty from the inline function probably pales compared to that of using resize-detector repeatedly, so I'll try to profile this and get back to you.

About using a custom HOC, that's a great idea! Also much cleaner than having it in `applyContainerQuery`.

```javascript
const customContainerQuery = (_baseQuery) => (Component) => ({ transformQuery, ...rest }) =>
<ContainerQuery query={transformQuery(_baseQuery)}>
  {(params) => <Component containerQuery={params} {...rest} />;}
</ContainerQuery>;
```
Works beautifully and it's good for #60

Reply:
BTW, I have fixed the lacking `componentWillReceiveProps` bug in v0.10 :D